MovingBlocks/Terasology
# maint: upgrade gradle to 6.4 from 6.0

This updates Gradle from 6.0 to the current release 6.4.1.

It fixes all gradle's deprecation warnings during build, and cleans up a few debugging print statements.

It also includes a [new groovy-wrapper.jar](https://github.com/MovingBlocks/groovy-wrapper/pull/4).

This is intended to be the “safe” parts of the work that started in #3959.

I don't think we have anything blocked waiting for gradle 6.4, but using current gradle provides a better baseline for other build streamlining work. Also brings it even with Launcher.


### How to test

- make sure your local dev builds still work
- also make sure we can still build releases


### Outstanding before Merge

- [x] ~merge #3954 (.idea), this branch is based on it.~
